gym/f110_gym/envs/action.py

```diff
@@ -19,73 +19,134 @@ def from_string(action: str):
-        self._action_type = "accl"
+        self._type = "speed"
+        self.lower_limit, self.upper_limit = params["v_min"], params["v_max"]
 
-        self.steering_low, self.steering_high = params["sv_min"], params["sv_max"]
-        self.acc_low, self.acc_high = -params["a_max"], params["a_max"]
+    def act(
+        self, action: Tuple[float, float], state: np.ndarray, params: Dict
+    ) -> float:
+        accl = pid_accl(
+            action,
+            state[3],
+            params["a_max"],
+            params["v_max"],
+            params["v_min"],
+        )
 
-    def act(self, action: Tuple[float, float], state, params) -> Tuple[float, float]:
-        return action
+        return accl
 
-    @property
-    def space(self) -> gym.Space:
-        low = np.array([self.steering_low, self.acc_low]).astype(np.float32)
-        high = np.array([self.steering_high, self.acc_high]).astype(np.float32)
+class SteerAction:
+    def __init__(self) -> None:
+        self._type = None
 
-        return gym.spaces.Box(low=low, high=high, shape=(2,), dtype=np.float32)
+        self.lower_limit = None
+        self.upper_limit = None
+
+    @abstractmethod
+    def act(self, steer_action: Any, **kwargs) -> float:
+        raise NotImplementedError("steer act method not implemented")
 
+    @property
+    def type(self) -> str:
+        return self._type
+    
+    @property
+    def space(self) -> gym.Space:
+        return gym.spaces.Box(low=self.lower_limit, high=self.upper_limit, dtype=np.float32)
 
-class SpeedAction(CarAction):
+class SteeringAngleAction(SteerAction):
     def __init__(self, params: Dict) -> None:
         super().__init__()
-        self._action_type = "speed"
-
-        self.steering_low, self.steering_high = params["s_min"], params["s_max"]
-        self.velocity_low, self.velocity_high = params["v_min"], params["v_max"]
+        self._type = "speed"
+        self.lower_limit, self.upper_limit = params["sv_min"], params["sv_max"]
 
     def act(
         self, action: Tuple[float, float], state: np.ndarray, params: Dict
-    ) -> Tuple[float, float]:
-        accl, sv = pid(
-            action[0],
-            action[1],
-            state[3],
+    ) -> float: 
+        sv = pid_steer(
+            action,
             state[2],
             params["sv_max"],
-            params["a_max"],
-            params["v_max"],
-            params["v_min"],
         )
-        return accl, sv
+        return sv
+    
+class SteeringSpeedAction(SteerAction):
+    def __init__(self, params: Dict) -> None:
+        super().__init__()
+        self._type = "speed"
+        self.lower_limit, self.upper_limit = params["s_min"], params["s_max"]
+
+    def act(
+        self, action: Tuple[float, float], state: np.ndarray, params: Dict
+    ) -> float: 
+        return action
+
+class SteerActionEnum(Enum):
+    Angle = 1
+    Speed = 2
+
+    @staticmethod
+    def from_string(action: str):
+        if action == "angle":
```

Comment:
again, should specifiy steering_angle and steering_speed, just to be less confusing